### ISSUE TITLE
Deletion of resource which doesn't exists breaks the pod

### ISSUE THREAD

**Issue by sm43** [manifestival/manifestival#92]
in [tektoncd/operator](https://github.com/tektoncd/operator), we have CRDs and CRs which we create during installation.
during uninstallation, we delete CRDs first and then rest of the resources. there is a time when we delete those CRs.
As the CRDs are already deleted it returns an error but the pod breaks here
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11e3438]

goroutine 414 [running]:
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetKind(...)
	/go/src/github.com/tektoncd/operator/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go:226
github.com/manifestival/manifestival.okToDelete(0x0)
	/go/src/github.com/tektoncd/operator/vendor/github.com/manifestival/manifestival/manifestival.go:234 +0x38
github.com/manifestival/manifestival.Manifest.delete({{0xc000667000, 0x2b, 0x40}, {0x1e9c350, 0xc000418000}, {{0x1ebe080, 0xc0008cb890}, 0x0}}, 0x1e9c350, {0x0, ...})
	/go/src/github.com/tektoncd/operator/vendor/github.com/manifestival/manifestival/manifestival.go:179 +0xae
github.com/manifestival/manifestival.Manifest.Delete({{0xc000667000, 0x2b, 0x40}, {0x1e9c350, 0xc000418000}, {{0x1ebe080, 0xc0008cb890}, 0x0}}, {0x0, 0x0, ...})
```

because in deletion manifestival client here
https://github.com/tektoncd/operator/blob/main/vendor/github.com/manifestival/manifestival/manifestival.go#L175
doesn't return error and tries to check a nil object which breaks the pod

/kind bug

**Comment by jcrossley3** [comment]
Thanks for the report! It does seem like that logic is flawed, since we shouldn't be calling `okToDelete` when `err != nil`.

I'm not a little surprised you're not seeing other problems after deleting CRD's before their CR's, but I guess k8s handles that gracefully?

**Comment by jcrossley3** [comment]
Just so I'm clear: do you expect the delete to return an error? or fail silently?

**Comment by sm43** [comment]
when we delete CRDs, all the CRs are cleaned up by kube 
and then if we deleted a CR it returns an error `the server could not find the requested resource`

> Just so I'm clear: do you expect the delete to return an error? or fail silently?

I am okay with returning an error. for our use case we do it during uninstallation, currently I have added code
to call `delete` after checking that resource exist.  
its just shouldn't break, any way is fine! :)
